src/providers/node/spa/vite.rs

```diff
@@ -0,0 +1,64 @@
+use regex::Regex;
+
+use crate::{nixpacks::app::App, providers::node::PackageJson};
+
+pub struct ViteSpaProvider {}
+
+impl ViteSpaProvider {
+    pub fn is_vite(app: &App) -> bool {
+        let package_json: PackageJson = app.read_json("package.json").unwrap_or_default();
+        package_json.has_dependency("vite")
+            || app.includes_file("vite.config.js")
+            || app.includes_file("vite.config.ts")
+            || {
+                let pkg: PackageJson = app.read_json("package.json").unwrap_or_default();
+                if let Some(scripts) = pkg.scripts {
+                    if let Some(build) = scripts.get("build") {
+                        build.to_lowercase().contains("vite build")
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                }
+            }
+    }
+
+    pub fn caddy_allowlist(app: &App) -> bool {
+        let pkg: PackageJson = app.read_json("package.json").unwrap();
+        pkg.has_dependency("react")
+            || pkg.has_dependency("vue")
+            || (pkg.has_dependency("svelte") && !pkg.has_dependency("@sveltejs/kit"))
+            || pkg.has_dependency("preact")
+            || pkg.has_dependency("lit")
+            || pkg.has_dependency("solid-js")
+            || pkg.has_dependency("@builder.io/qwik")
+    }
+
+    pub fn get_output_directory(app: &App) -> String {
+        let config = app
+            .read_file("vite.config.js")
+            .or(app.read_file("vite.config.ts"));
+        let r = Regex::new(r#"outDir:\s*['"`](.*?)['"`]"#).unwrap();
```

Comment:
Can you add some tests in this file for the regex and these checks. I'm a little concerned of accidentally considering apps as SPAs when the shouldn't be.